src/Error.cc

```diff
@@ -118,6 +120,19 @@ void Error::SetXmlPath(const std::string &_xmlPath)
   this->dataPtr->xmlPath = _xmlPath;
 }
 
+/////////////////////////////////////////////////
+void Error::throwOrPrintError()
```

Comment:
I think it would be a good idea for this to take an `ostream` object as input so it can be used with `sdfwarn` and others.

Reply:
`sdferr` and `sdfwarn` seem to be `ConsoleStream` shall it take that one as input instead of the `ostream`?

Reply:
Added a `ConsoleStream` for now in https://github.com/gazebosim/sdformat/pull/1220/commits/6483e3ab20de09fc28f83e16ed45a67297419466. Let me know if we should do something different.

Reply:
Yeah, I thought `ConsoleStream` inherited from `ostream`, but that's not the case, so what you have is good.